Bug: prefer-arrow-callback autofix breaks async + multi-line comment

### Environment

- ESLint version: `v10.4.0`
- Node.js version: `v24.15.0`
- npm version: `11.12.1`
- Operating System: Windows 11 

### What parser are you using?

Default (Espree)

### What did you do?

Running `--fix` on an async function callback that has a multi-line block comment between `function` and the parameter list (`repro.js`):

```js
foo(async function /*
*/ () { return 1; });
```

Config:

```json
{ "rules": { "prefer-arrow-callback": ["error"] } }
```

Command (local clone at commit `9067f9492`):



### What did you expect to happen?

The autofix should give back valid JavaScript. If it can't do that safely (like in this case), it should skip the fix and just report the warning instead.

### What actually happened?



The autofix turned `repro.js` into this:

```js
foo(async  /*
*/ () => { return 1; });
```

This is broken — it doesn't parse anymore. The ESLint homepage says "ESLint fixes are syntax-aware so you won't experience errors introduced by traditional find-and-replace algorithms." but that's exactly what happened here.

The failure shows up in two ways:

**1. ESLint itself reports a parse error on the same `--fix` run (exit code 1):**

```
repro.js
  2:7  error  Parsing error: Unexpected token =>

✖ 1 problem (1 error, 0 warnings)
```

**2. Espree also chokes on the post-fix file:**

```bash
node -e "require('espree').parse(require('fs').readFileSync('repro.js','utf8'),{ecmaVersion:'latest',sourceType:'module'})"
```

```
SyntaxError: Unexpected token => (2:7)
```

perfectly valid JS to something that can't even be parsed.






### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiZm9vKGFzeW5jIGZ1bmN0aW9uIC8qXG4qLyAoKSB7IHJldHVybiAxOyB9KTsiLCJvcHRpb25zIjp7InJ1bGVzIjp7InByZWZlci1hcnJvdy1jYWxsYmFjayI6WyJlcnJvciJdfSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=

### Participation

- [x] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [ ] I did not use AI to generate this issue report.
- [x] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

**Why the post-fix output is invalid (spec reference)**

The `AsyncArrowFunction` production in the spec says:

> `async [no LineTerminator here] AsyncArrowBindingIdentifier [no LineTerminator here] => AsyncConciseBody`

The `CoverCallExpressionAndAsyncArrowHead` variant has the same `[no LineTerminator here]` between `async` and the arrow head. A block comment that has `\n` in it counts as a LineTerminator for this rule. In the fixed output, the comment `/* \n */` sits between `async` and `()`, so it puts a line break where one isn't allowed.

Spec link: https://tc39.es/ecma262/#prod-AsyncArrowFunction

--- a/lib/rules/prefer-arrow-callback.js
+++ b/lib/rules/prefer-arrow-callback.js
@@ -333,7 +333,6 @@ module.exports = {
 							);
 
 							if (node.async) {
-								
 								if (
 									functionToken.loc.end.line <
 									leftParenToken.loc.start.line

--- a/lib/rules/prefer-arrow-callback.js
+++ b/lib/rules/prefer-arrow-callback.js
@@ -333,13 +333,10 @@ module.exports = {
 							);
 
 							if (node.async) {
-								const textBetween = sourceCode.text.slice(
-									functionToken.range[1],
-									leftParenToken.range[0],
-								);
-
+								
 								if (
-									astUtils.LINEBREAK_MATCHER.test(textBetween)
+									functionToken.loc.end.line <
+									leftParenToken.loc.start.line
 								) {
 									return;
 								}

--- a/lib/rules/prefer-arrow-callback.js
+++ b/lib/rules/prefer-arrow-callback.js
@@ -323,6 +323,28 @@ module.exports = {
 								return;
 							}
 
+							const functionToken = sourceCode.getFirstToken(
+								node,
+								node.async ? 1 : 0,
+							);
+							const leftParenToken = sourceCode.getTokenAfter(
+								functionToken,
+								astUtils.isOpeningParenToken,
+							);
+
+							if (node.async) {
+								const textBetween = sourceCode.text.slice(
+									functionToken.range[1],
+									leftParenToken.range[0],
+								);
+
+								if (
+									astUtils.LINEBREAK_MATCHER.test(textBetween)
+								) {
+									return;
+								}
+							}
+
 							// Remove `.bind(this)` if exists.
 							if (callbackInfo.isLexicalThis) {
 								const memberNode = node.parent;
@@ -375,14 +397,6 @@ module.exports = {
 							}
 
 							// Convert the function expression to an arrow function.
-							const functionToken = sourceCode.getFirstToken(
-								node,
-								node.async ? 1 : 0,
-							);
-							const leftParenToken = sourceCode.getTokenAfter(
-								functionToken,
-								astUtils.isOpeningParenToken,
-							);
 							const tokenBeforeBody = sourceCode.getTokenBefore(
 								node.body,
 							);

--- a/tests/lib/rules/prefer-arrow-callback.js
+++ b/tests/lib/rules/prefer-arrow-callback.js
@@ -181,6 +181,26 @@ ruleTester.run("prefer-arrow-callback", rule, {
 			errors,
 		},
 		{
+			code: "foo(async function /*\n*/ () { return 1; });",
+			output: null,
+			errors,
+		},
+		{
+			code: "foo(async function // c\n () { return 1; });",
+			output: null,
+			errors,
+		},
+		{
+			code: "foo(async function\n () { return 1; });",
+			output: null,
+			errors,
+		},
+		{
+			code: "foo(async function /* c */ () { return 1; });",
+			output: "foo(async  /* c */ () => { return 1; });",
+			errors,
+		},
+		{
 			code: "foo((bar || function() {}).bind(this))",
 			output: null,
 			errors,

